includes/functions.php

```diff
@@ -76,3 +76,136 @@ function get_stories_theme_support() {
 
 	return $theme_support;
 }
+
+/**
+ * Wrapper function for fetching field states
+ * based on the view types.
+ *
+ * Mainly uses FieldState and Fields classes.
+ *
+ * @SuppressWarnings(PHPMD.UnusedLocalVariable)
+ *
+ * @return array
+ */
+function fields_states() {
+	$views = get_layouts();
+
+	$fields = [
+		'title',
+		'author',
+		'date',
+		'image_align',
+		'excerpt',
+		'archive_link',
+	];
+
+	$field_states = [];
+
+	foreach ( $views as $view_type => $view_label ) {
+		$field_state = ( new Story_Query( [ 'view_type' => $view_type ] ) )->get_renderer()->field();
+		foreach ( $fields as $field ) {
+			$field_states[ $view_type ][ $field ] = [
+				'show'     => $field_state->$field()->show(),
+				'label'    => $field_state->$field()->label(),
+				'readonly' => $field_state->$field()->readonly(),
+			];
+		}
+	}
+
+	return $field_states;
+}
+
+/**
+ * Put some tinymce related data on the page.
+ *
+ * @param string $hook Hook name for current page.
+ *
+ * @return void
+ */
+function web_stories_script_data( $hook ) {
+	if ( 'widgets.php' !== $hook ) {
+		return;
+	}
+
+	$order      = get_orderby();
+	$views      = get_layouts();
+	$order_list = [];
+	$view_types = [];
+
+	foreach ( $order as $order_key => $an_order ) {
+		$order_list[] = [
+			'label' => $an_order,
+			'value' => $order_key,
+		];
+	}
+
+	foreach ( $views as $view_key => $view_label ) {
+		$view_types[] = [
+			'label' => $view_label,
+			'value' => $view_key,
+		];
+	}
+
+	$field_states = fields_states();
+
+	$data = [
+		'orderlist' => $order_list,
+		'icon'      => WEBSTORIES_PLUGIN_DIR_URL . '/__static__/carousel-icon.png',
```

Comment:
Assets should live in `assets/images/`

Reply:
Yup. `__static__` is only for testing, and not part of the final plugin ZIP file.

Reply:
Moved it to `assets/images/`